subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -666,3 +667,29 @@ function getRepayForLiquidation(event: ethereum.Event): BigInt | null {
 
   return null;
 }
+
+export function updateWeightedBorrow(mrkt: Market, epoch: _Epoch, endBlock: BigInt): void {
+  // update total borrow balance USD weighted by blocks lapsed
+  const mktReceivingRewards = mrkt._receivingRewards;
+  log.debug("[updateWeightedBorrow]mrkt {} _receivingRewards={}", [mrkt.id, mktReceivingRewards.toString()]);
+  if (mrkt._receivingRewards) {
+    if (!mrkt._borrowLastUpdateBlock) {
+      const epochStartBlock = getStartBlockForEpoch(epoch.epoch)!;
+      mrkt._borrowLastUpdateBlock = epochStartBlock;
+      mrkt._weightedTotalBorrowUSD = BIGDECIMAL_ZERO;
+    }
+
+    const blocksLapsed = endBlock.minus(mrkt._borrowLastUpdateBlock!);
+    const addWeightedTotalBorrowUSD = mrkt.totalBorrowBalanceUSD.times(blocksLapsed.toBigDecimal());
+    mrkt._weightedTotalBorrowUSD = mrkt._weightedTotalBorrowUSD!.plus(addWeightedTotalBorrowUSD);
+    mrkt._borrowLastUpdateBlock = endBlock;
+    mrkt.save();
+  }
+}
+
+export function updateWeightedStakeAmount(market: Market, endBlock: BigInt): void {
+  const blocksLapsed = endBlock.minus(market._stakeLastUpdateBlock!);
```

Comment:
does `market._stakeLastUpdateBlock` need to be updated after this function?

Reply:
Good point! Currently, it is done after the function call (needed before a refactoring), but it can be moved inside the function.

Reply:
Would increase readability that way (for me at least)